### PR TITLE
[MNT] Added codespell to github action workflow

### LINKED ISSUE #33: Add codespell to github action workflow

### PR DESCRIPTION
**Pull request by Raya679**
- Closes #33 

Changes proposed in this pull request:

- Added codespell to github action workflow

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/workflows/codespell.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        
+      - name: Codespell
+        uses: codespell-project/actions-codespell@v2
+        with: 
+          skip: ./app/term_url_processing/abbreviations/abbreviations_sex.py,./app/term_url_processing/abbreviations/abbreviations_diagnosis.py,./app/term_url_processing/abbreviations/abbreviations_assessment.py,./app/term_url_processing/abbreviations/abbreviations_image_modality.py
```
Comment (rmanaem) [inline .github/workflows/codespell.yaml]:
```suggestion
          skip: ./app/term_url_processing/abbreviations/*
```

**Review by rmanaem — APPROVED** [review]
Thanks for the PR @Raya679! 
I left one suggestion other than that this is good to go 🧑‍🍳

**Comment by Raya679** [comment]
Hii @rmanaem, 
I tried using ` skip: ./app/term_url_processing/abbreviations/*` but i think it is not able to recognize that. 
Should I go back to how it was previously?

**Comment by rmanaem** [comment]
Looking at their docs I think removing the wild card should do it: https://github.com/codespell-project/codespell?tab=readme-ov-file#usage

try `skip: ./app/term_url_processing/abbreviations`

**Pull request merged**